presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -233,6 +240,32 @@ public boolean isLimitGuaranteed(ConnectorSession session)
         return true;
     }
 
+    @Override
+    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        return format("CREATE TABLE %s (%s) WITH (DATA_COMPRESSION = %s)", quoted(remoteTableName), join(", ", columns), getDataCompression(tableMetadata.getProperties()));
+    }
+
+    @Override
+    public Map<String, Object> getTableProperties(JdbcIdentity identity, JdbcTableHandle tableHandle)
+    {
+        ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+
+        try (Connection connection = connectionFactory.openConnection(identity);
+                PreparedStatement statement = connection.prepareStatement("SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id WHERE p.partition_number = 1 AND t.name = ?")) {
+            statement.setString(1, tableHandle.getRemoteTableName().getTableName());
+            try (ResultSet resultSet = statement.executeQuery()) {
```

Comment:
jdbi3?

Reply:
But for using JDBI we might need inject a `ConnectionProvider` but for fetching a connection we need to `JdbcIdentity`

Reply:
i didn't quite get it. Anyway, you can attach jdbi to an existing connector.